### PR TITLE
Feat/gw7652 remove emoji auto complete helper

### PR DESCRIPTION
**Pull request by mudana-grune**
https://youtrack.weseek.co.jp/issue/GW-7652
- Remove implementation of EmojiAutoCompleteHelper

### CODE REVIEW

File: packages/app/src/components/PageEditor/CodeMirrorEditor.jsx [hunk 1]
```diff
@@ -932,24 +944,42 @@ export default class CodeMirrorEditor extends AbstractEditor {
...
+        clearTimeout(this.state.searchEmojiTimeout);
+      }
+      const timeout = setTimeout(() => {
+        const currentSearchText = sc.matches(true, pos).match[0];
+        const searchValue = currentSearchText.replace(':', '');
         this.setState({ isEmojiPickerShown: isInputtingEmoji });
         this.setState({ emojiSearchText: searchValue });
-      }, 1000);
+      }, 700);
+      this.setState({ searchEmojiTimeout: timeout });
```
[the remark this answers is not in the document]
Comment (mudana-grune) [reply packages/app/src/components/PageEditor/CodeMirrorEditor.jsx:982]: This timeout actually aims to detect if the user stops typing at a certain time. If the emoji picker appears instantly, maybe the user can only type 1 letter. Please give me suggestions for this.

[the remark this answers is not in the document]
Comment (kaoritokashiki) [reply packages/app/src/components/PageEditor/CodeMirrorEditor.jsx:982]: なるほどです。
https://user-images.githubusercontent.com/59536731/152095871-a28a0753-0ce9-4d93-b04d-399d45bd9f92.mov

File: packages/app/src/components/PageEditor/EmojiPicker.jsx [hunk 2]
```diff
@@ -9,6 +9,58 @@ class EmojiPicker extends React.Component {
     super(props);
     this.emojiPicker = React.createRef();
     this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.selectEmoji = this.selectEmoji.bind(this);
+  }
```
Comment (yuki-takei) [inline packages/app/src/components/PageEditor/EmojiPicker.jsx:13]: Convert to a functional component please.

File: packages/app/src/components/PageEditor/CodeMirrorEditor.jsx [hunk 3]
```diff
@@ -665,16 +665,42 @@ export default class CodeMirrorEditor extends AbstractEditor {
...
+      sc.replace(emoji.colons, cm.getTokenAt(currentPos).string);
+      this.setState({ emojiSearchText: null });
+      cm.focus();
+    }
+    else {
+      doc.replaceRange(emoji.colons, currentPos);
+      cm.focus();
+    }
+
+  }
```
Comment (yuki-takei) [inline packages/app/src/components/PageEditor/CodeMirrorEditor.jsx:691]: Transplant to an utility module like EmojiAutoCompleteHelper please.

I think it's better not to write such emoji specific processes to CodeMirrorEditor component.

**Review by yuki-takei — CHANGES_REQUESTED** [review]

File: packages/app/src/components/PageEditor/CodeMirrorEditor.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuki-takei) [reply packages/app/src/components/PageEditor/CodeMirrorEditor.jsx:982]: これは一旦OKです

File: packages/app/resource/locales/ja_JP/translation.json [hunk 4]
```diff
@@ -936,5 +936,35 @@
...
+    },
+    "categorieslabel": "Emoji categories",
+    "skintones": {
+      "1": "Default Skin Tone",
+      "2": "Light Skin Tone",
+      "3": "Medium-Light Skin Tone",
+      "4": "Medium Skin Tone",
+      "5": "Medium-Dark Skin Tone",
+      "6": "Dark Skin Tone"
+    }
```
Comment (yuki-takei) [inline packages/app/resource/locales/ja_JP/translation.json:968]: Please fill these value in Japanese with the help of others.

File: packages/app/resource/locales/zh_CN/translation.json [hunk 5]
```diff
@@ -946,5 +946,35 @@
...
+    },
+    "categorieslabel": "Emoji categories",
+    "skintones": {
+      "1": "Default Skin Tone",
+      "2": "Light Skin Tone",
+      "3": "Medium-Light Skin Tone",
+      "4": "Medium Skin Tone",
+      "5": "Medium-Dark Skin Tone",
+      "6": "Dark Skin Tone"
+    }
```
Comment (yuki-takei) [inline packages/app/resource/locales/zh_CN/translation.json:978]: Please fill these value in Chinese. It's enough to use machine translation.

File: packages/app/src/components/PageEditor/Emoji.jsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+Emoji.propTypes = {
+  t: PropTypes.func.isRequired,
+  close: PropTypes.func,
+  selectEmoji: PropTypes.func,
+  emojiSearchText: PropTypes.string,
+};
+
+export default withTranslation()(Emoji);
```
Comment (yuki-takei) [inline packages/app/src/components/PageEditor/Emoji.jsx:101]: - Unify this file with EmojiPicker.jsx
- Convert to TypeScript (tsx)
- Convert to a Functional Component

File: packages/app/src/components/PageEditor/EmojiPickerHelper.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    return;
+
+  }
+
+
+}
+
+export default EmojiPickerHelper;
```
Comment (yuki-takei) [inline packages/app/src/components/PageEditor/EmojiPickerHelper.js:64]: - Convert to TypeScript

**Review by yuki-takei — CHANGES_REQUESTED** [review]

**Pull request merged**